From ce6abc4e8c37d653b61e00df115ffe219e63f6d7 Mon Sep 17 00:00:00 2001
Message-Id: <ce6abc4e8c37d653b61e00df115ffe219e63f6d7.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Sat, 5 Jul 2014 01:53:22 +0530
Subject: [PATCH 097/312] REVERTME [VPG]: Correct the display detection
 ordering on VLV

On VLV boards, the HDMI is on Port B and eDP and MIPI on Port C.
Display detection and init order needs to be taken care.
This is a REVERTME. Corrected ordering will be pushed later.

Issue: GMIN-1475
Change-Id: I6e3da9bd73e37734df08f113567a49b66c158928
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |   36 +++++++++++++++++-----------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 1f30078..eb6b267 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12181,30 +12181,30 @@ static void intel_setup_outputs(struct drm_device *dev)
 		if (I915_READ(PCH_DP_D) & DP_DETECTED)
 			intel_dp_init(dev, PCH_DP_D, PORT_D);
 	} else if (IS_VALLEYVIEW(dev)) {
-		if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIB) & SDVO_DETECTED) {
-			intel_hdmi_init(dev, VLV_DISPLAY_BASE + GEN4_HDMIB,
-					PORT_B);
-			if (I915_READ(VLV_DISPLAY_BASE + DP_B) & DP_DETECTED)
-				intel_dp_init(dev, VLV_DISPLAY_BASE + DP_B, PORT_B);
-		}
-
-		if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIC) & SDVO_DETECTED) {
-			intel_hdmi_init(dev, VLV_DISPLAY_BASE + GEN4_HDMIC,
-					PORT_C);
-			if (I915_READ(VLV_DISPLAY_BASE + DP_C) & DP_DETECTED)
+		/* There is no detection method for MIPI so rely on VBT */
+		if (!IS_CHERRYVIEW(dev)) {
+			if (dev_priv->vbt.has_mipi)
+				intel_dsi_init(dev);
+			else if (I915_READ(VLV_DISPLAY_BASE + DP_C) &
+					DP_DETECTED)
 				intel_dp_init(dev, VLV_DISPLAY_BASE + DP_C, PORT_C);
-		}
-
-		if (IS_CHERRYVIEW(dev)) {
-			if (I915_READ(VLV_DISPLAY_BASE + CHV_HDMID) & SDVO_DETECTED) {
+			if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIB) &
+					SDVO_DETECTED) {
+				intel_hdmi_init(dev,
+					VLV_DISPLAY_BASE + GEN4_HDMIB,
+					PORT_B);
+			}
+		} else if (IS_CHERRYVIEW(dev)) {
+			if (I915_READ(VLV_DISPLAY_BASE + CHV_HDMID) &
+					SDVO_DETECTED) {
 				intel_hdmi_init(dev, VLV_DISPLAY_BASE + CHV_HDMID,
 						PORT_D);
-				if (I915_READ(VLV_DISPLAY_BASE + DP_D) & DP_DETECTED)
+				if (I915_READ(VLV_DISPLAY_BASE + DP_D) &
+						DP_DETECTED)
 					intel_dp_init(dev, VLV_DISPLAY_BASE + DP_D, PORT_D);
 			}
+			intel_dsi_init(dev);
 		}
-
-		intel_dsi_init(dev);
 	} else if (SUPPORTS_DIGITAL_OUTPUTS(dev)) {
 		bool found = false;
 
-- 
1.7.9.5

